### DISCUSSION TITLE
(beginner) optimize seems to return a wrong result if I specify the objective.

### DISCUSSION THREAD

**Discussion by ZigZagCrickCrack** [yalmip/YALMIP#1251] [Problems using YALMIP]
Hi @johanlofberg sorry to bother you,

I'm 3 or 4 hours into learning to use Yalmip, I am trying to write a code that calculates L2 gain of a simple example system, just like you showed in the "beginner mistakes" article on the yalmip website (and therefore I feel shit for having to come here). 
If I write the problem like so, which means that I don't specify the objective function, it seems to find the right value of the gain.

```
A = [0 1; -1 -0.02];
B = [0;1];
C = [1 0];

sys = ss(A,B,C,0)

bode(sys)
grid on

yalmip('clear')

P = sdpvar(2, 2)
t = sdpvar(1)

F = [P>=0, [P*A + A'*P + C'*C, P*B; B'*P, -t]<=0, t>=0]
diagnostics = optimize(F)

P = double(value(P))
gamma = sqrt(value(t))

```
while if with the same exact problem I specify that i want to minimize t:

`diagnostics = optimize(F, t)`

the solver returns 0 as if the problem was solved, but the value of the gain is actually wrong, and running the command check(F) I get a negative residual on the LMI:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
|   ID|               Constraint|   Primal residual|   Dual residual|
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
|   #1|        Matrix inequality|           49.5051|      7.4211e-09|
|   #2|        Matrix inequality|       -2.4496e-10|      9.4479e-11|
|   #3|   Elementwise inequality|           2500.25|      9.4087e-11|
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

- what is going on?
- Not specifying the objective function implies that all the decision variables get minimized?

Thanks for the help!

**Comment by ZigZagCrickCrack** [comment]
Update: after double checking the bode plot actually the second result goes closer to the gain that i expected, but it still remains that there is a negative primal residual and the fact that the two results are different.
I was more convinced that the first was the right one also because that is basically the script that was used by my professor.

**Reply by johanlofberg** [reply]
In the first you only solve a feasibility problem, while in the second you actually optimize, so why would you expect anything close? The first will not get you the L2 gain (it will get you some upper bound on the gain)

you can almost never expect feasible solutions when you optimize
https://yalmip.github.io/faq/solutionviolated/

**Reply by ZigZagCrickCrack** [reply]
Yeah I think I got confused by my professor's script. Since it was in an exercise about calculating the L2 gain, but apparently the code simply did not do that.
Thanks a lot, you are awesome.
